framework/include/utils/MooseTypes.h

```diff
@@ -231,6 +231,13 @@ enum SolutionState : int
   Older = 2,
   PreviousNL = -1
 };
+
+enum class SolutionIterationType : unsigned short
+{
+  Time = 0,
+  Nonlinear
```

Comment:
do you envision linear and fixedPoint coming in soon or probably not

Reply:
I don't envision linear ever coming in, but my vision is not perfect. I could see fixed point coming in. But I didn't want to guess at enumeration cases until there is a use case